nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py

```diff
@@ -210,72 +174,116 @@ def __init__(
+                layernorm_epsilon=decoder_cfg.layernorm_epsilon,
+                bias_activation_fusion=decoder_cfg.get('bias_activation_fusion', True),
+                bias_dropout_add_fusion=decoder_cfg.get('bias_dropout_add_fusion', True),
+                masked_softmax_fusion=decoder_cfg.get('masked_softmax_fusion', True),
+                persist_layer_norm=decoder_cfg.get('persist_layer_norm', True),
+                openai_gelu=decoder_cfg.get('openai_gelu', False),
+                onnx_safe=decoder_cfg.get('onnx_safe', False),
+                hidden_steps=decoder_cfg.get('hidden_steps', -1),
+                activation=decoder_cfg.get('activation', 'gelu'),
+                bias=decoder_cfg.get('bias', True),
+                normalization=decoder_cfg.get('normalization', 'layernorm'),
+                transformer_block_type=decoder_cfg.get('transformer_block_type', 'pre_ln'),
+                headscale=decoder_cfg.get('headscale', False),
                 parent_model_type=ModelType.encoder_and_decoder,
             )
 
         self.enc_dec_model = MegatronTransformerEncoderDecoderModule(
-            encoder=encoder, decoder=decoder, hidden_steps=hidden_steps
+            encoder=encoder, decoder=decoder, hidden_steps=encoder_cfg.get('hidden_steps', -1),
         )
         self._enc_dec_model_key = "enc_dec_model"
 
         self.initialize_word_embeddings(
-            init_method=init_method_normal(init_method_std), vocab_size=vocab_size, hidden_size=hidden_size
+            init_method=init_method_normal(embedding_init_method_std),
+            vocab_size=vocab_size,
+            hidden_size=encoder_cfg.hidden_size,
         )
 
         if add_decoder and post_process:
             self.tokens_head = MegatronTokenLevelHead(self.word_embeddings_weight().size(0), parallel_output)
             self._tokens_head_key = 'tokens_head'
 
+    def _validate_kv_channels(self, cfg):
+        kv_channels = cfg.kv_channels
+        if cfg.kv_channels is None:
+            assert (
+                cfg.hidden_size % cfg.num_attention_heads == 0
+            ), 'hidden_size must be divisible by num_attention_heads if kv_channels is None'
+            kv_channels = cfg.hidden_size // cfg.num_attention_heads
+
+        return kv_channels
+
+    def _validate_enc_dec_hidden_size(self, encoder_cfg, decoder_cfg):
+        if encoder_cfg.hidden_size != decoder_cfg.hidden_size:
+            raise ValueError(
+                f"Encoder and decoder hidden_size must be equal, but got encoder: {encoder_cfg.hidden_size} and decoder: {decoder_cfg.hidden_size}"
+            )
+
+    # TODO: (sandeepsub) - this can be removed one we rebase to sequence parallel that contains the RPE refactor.
+    def _validate_position_embedding_type(self, encoder_cfg, decoder_cfg):
+        if encoder_cfg.get('position_embedding_type', 'learned_absolute') != decoder_cfg.get(
```

Comment:
Why positional embeddings of encoder and decoder must be the same? Not required if untied (make sure to update when merging to untied embeddings PR).

Reply:
Done! I no longer need to validate position embedding types.